x-pack/test/detection_engine_api_integration/security_and_spaces/tests/perform_bulk_action.ts

```diff
@@ -156,5 +145,51 @@ export default ({ getService }: FtrProviderContext): void => {
 
       expect(rulesResponse.total).to.eql(2);
     });
+
+    it('should set, add and delete tags in rules', async () => {
```

Comment:
Should we add tests for index and timeline actions as well?

Reply:
I have it covered in unit tests: https://github.com/elastic/kibana/blob/61efff8ee88e2a23d98a6618be0e24e6a4d070a5/x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_action_update.test.ts#L74

Thought, it might be redundant to do it for functional tests, as I already covered update action(for tags) to confirm e2e running is successful. 
Do you think we need to a functional test for every action?

Reply:
Unit tests for the `appplyBulkActionUpdateToRule` method are good, but they don't cover other methods used by the bulk action API. For example, if something changes inside the `patchRules` method, how would it affect our bulk API? Unit tests cannot answer that question, as they test methods in isolation. On the other hand, functional tests test API as a whole and verify that we should expect to receive particular output for the given input no matter what changes inside the route. And that's what we need in this case.

Reply:
added additional test